### ISSUE TITLE
[BUG] Boolean filter creates invalid OData string

### ISSUE THREAD

**Issue by cheng93** [Azure/azure-sdk-for-net#30185]
### Library name and version

Azure.Data.Tables 12.6.1

### Describe the bug

```csharp
table.QueryAsync<Entity>(x => x.Enabled);
```

creates an invalid OData url
`[28/Jul/2022:10:27:11 +0000] "GET /devstoreaccount1/privatedefinitions()?$format=application%2Fjson%3Bodata%3Dminimalmetadata&$filter=Enabled HTTP/1.1" 400`

```csharp
table.QueryAsync<Entity>(x => x.Enabled == true);
```

is valid
`[28/Jul/2022:10:31:01 +0000] "GET /devstoreaccount1/privatedefinitions()?$format=application%2Fjson%3Bodata%3Dminimalmetadata&$filter=Enabled%20eq%20true HTTP/1.1" 200`

### Expected behavior

should produce a valid OData url

### Actual behavior

see above

### Reproduction Steps

see above

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by cheng93** [comment]
Closed in #30863
